linode/firewall/framework_models.go

```diff
@@ -0,0 +1,168 @@
+package firewall
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-framework/types/basetypes"
+	"github.com/linode/linodego"
+)
+
+// FirewallModel describes the Terraform resource data model to match the
+// resource schema.
+type FirewallModel struct {
+	ID             types.Int64  `tfsdk:"id"`
+	Label          types.String `tfsdk:"label"`
+	Tags           types.Set    `tfsdk:"tags"`
+	Disabled       types.Bool   `tfsdk:"disabled"`
+	Inbound        types.List   `tfsdk:"inbound"`
+	InboundPolicy  types.String `tfsdk:"inbound_policy"`
+	Outbound       types.List   `tfsdk:"outbound"`
+	OutboundPolicy types.String `tfsdk:"outbound_policy"`
+	Linodes        types.Set    `tfsdk:"linodes"`
+	Devices        types.List   `tfsdk:"devices"`
+	Status         types.String `tfsdk:"status"`
+}
+
+func (data *FirewallModel) parseComputedAttributes(
+	ctx context.Context,
+	firewall *linodego.Firewall,
+	rules *linodego.FirewallRuleSet,
+	devices []linodego.FirewallDevice,
+) diag.Diagnostics {
+	data.ID = types.Int64Value(int64(firewall.ID))
```

Comment:
I think ID isn't computed, right?

Reply:
ID will need to be computed for the resource. https://github.com/linode/terraform-provider-linode/blob/dev/linode/firewall/schema_resource.go#L97 Id is not included in the sdkv2 resource schema, however, when we migrate it to the framework, we have to explicitly add it in the resource schema and mark it computed.

Reply:
I think for we split the parse function into two parts, one for computed and another for non-computed attributes because in the resource, sometimes we don't want to edit the non-computed attributes defined by the resource schema. So I think wether a attribute should be placed in parsing computed or non-computed function is decided by the resource schema. You can take a look at the [stackscript module](https://github.com/linode/terraform-provider-linode/tree/dev/linode/stackscript) where we migrated both resource and data source. Although some of the attributes are computed in the data source schema and non-computed in resource schema, they are still in parse non-computed function because they are non-computed in the resource schema. 

For example, `label`

https://github.com/linode/terraform-provider-linode/blob/dev/linode/stackscript/framework_models.go#L40

https://github.com/linode/terraform-provider-linode/blob/dev/linode/stackscript/framework_datasource_schema.go#L15-L18

https://github.com/linode/terraform-provider-linode/blob/dev/linode/stackscript/framework_datasource_schema.go#L15-L18

Reply:
I see. We probably should consider rename these computed vs non-computed functions once we have a plan for the resource migration in the future. The current naming is a bit confusing.